monkey/monkey_island/cc/services/post_breach_files.py

```diff
@@ -1,53 +1,44 @@
 import logging
 import os
-
-import monkey_island.cc.services.config
+from pathlib import Path
 
 __author__ = "VakarisZ"
 
-import monkey_island.cc.environment.environment_singleton as env_singleton
-
 logger = logging.getLogger(__name__)
 
 # Where to find file names in config
-PBA_WINDOWS_FILENAME_PATH = ["monkey", "post_breach", "PBA_windows_filename"]
 PBA_LINUX_FILENAME_PATH = ["monkey", "post_breach", "PBA_linux_filename"]
+PBA_WINDOWS_FILENAME_PATH = ["monkey", "post_breach", "PBA_windows_filename"]
 
 
-def remove_PBA_files():
-    if monkey_island.cc.services.config.ConfigService.get_config():
-        windows_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_WINDOWS_FILENAME_PATH
-        )
-        linux_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_LINUX_FILENAME_PATH
-        )
-        if linux_filename:
-            remove_file(linux_filename)
-        if windows_filename:
-            remove_file(windows_filename)
-
-
-def remove_file(file_name):
-    file_path = os.path.join(env_singleton.env.get_config().data_dir_abs_path, file_name)
-    try:
-        if os.path.exists(file_path):
-            os.remove(file_path)
-    except OSError as e:
-        logger.error("Can't remove previously uploaded post breach files: %s" % e)
-
-
-def set_config_PBA_files(config_json):
-    """
-    Sets PBA file info in config_json to current config's PBA file info values.
-    :param config_json: config_json that will be modified
-    """
-    if monkey_island.cc.services.config.ConfigService.get_config():
-        linux_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_LINUX_FILENAME_PATH
-        )
-        windows_filename = monkey_island.cc.services.config.ConfigService.get_config_value(
-            PBA_WINDOWS_FILENAME_PATH
+class PostBreachFilesService:
+    DATA_DIR = None
```

Comment:
Why can't this point to the singleton value of data_dir? If we can do it, then there's no need for initialization

Reply:
We want to decouple this from the singleton. If changes are made to the environment config, we don't want those changes to ripple through the entire codebase. Passing this code only the information it needs to get its job done reduces coupling and makes it easier to test.

Furthermore, the next PR removes `data_dir` from the environment singleton.

Reply:
Except that PBA's are coupled with the data_dir which is and always will be a global param. What I don't like is that instead of having a one singleton to represent global state, now the global state is split and stored in class objects. I'm also not sure about initializing classes, this sounds a bit like an anti-pattern. Instead, I feel like we should decouple the singleton from the environment config. This would make it so that changes to environment config won't ripple through the codebase and would avoid awkward and unnecessary initialization just to read a global param.

Reply:
We're decoupling the `data_dir` from the environment config. See https://github.com/guardicore/monkey/pull/1161.